design_docs/selfhealing_featuretoggle.md

```diff
@@ -0,0 +1,160 @@
+
+
+# User Walkthrough
+
+
+## Setup & Configure Keptn 
+
+As you would normally do with Keptn, create a project, onboard a service, and deploy your service into your Kubernetes cluster.
+
+<details><summary>Instructions</summary>
+
+### Create project 
+
+Create a Keptn project which where you want to onboard your service. Define all stages you want to have in your project in a shipyard file, similar to the [Keptn docs](https://keptn.sh/docs/0.5.0/usecases/onboard-carts-service/#create-project-sockshop).
+
+```
+keptn create project PROJECTNAME --shipyard=shipyard.yaml
+```
+
+
+
+
+
+### Onboard a service
+
+Onboard the service and add the remediation file as a resource.
+
+
+```
+keptn onboard service SERVICENAME --project=PROJECTNAME --chart=HELMCHART
+```
+
+
+Configure Keptn - add the remediation file
+
+
+```
+keptn add-resource --project=PROJECTNAME --service=SERVICENAME --stage=STAGENAME --resource=remediation.yaml
+```
+
+Deploy your application to Keptn.
+
+```
+keptn send event new-artifact --project=PROJECTNAME --service=SERVICENAME --image=docker.io/IMAGENAME --tag=TAG
+```
+
+</details>
+
+
+### Add secret for Unleash server
+
+In order to let the Keptn built-in Unleash remediation service communicate with the Unleash server, provide the credentials in terms of a Kubernetes secret.
+
+unleash-secret.yaml:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: unleash-secret
+type: Opaque
+data:
+  username: unleash-username
+  token: unleash-token
+  server-url: unleash-server-url
+```
+
+
+Apply the secret to the cluster.
+
+```
+kubectl apply -f unleash-secret.yaml -n keptn
+```
+
+<!--
+ALTERNATIVE
+
+```
+#keptn add secret secret.yaml # needed to access the unleash server
+#keptn apply uniform --project=PROJECTNAME uniform.yaml
+```
+
+uniform not needed right now since unleash is built in as action in Keptn core remediation service 
+-->
+
+### Configure your monitoring
+
+To be able to react on any issues in your environment, the monitoring solution has to be setup and configured to send details of the detected issue to Keptn. In case of Dynatrace, this is how it would look like:
+
+1. Follow the Keptn docs to setup Dynatrace [as described here](https://keptn.sh/docs/0.5.0/reference/monitoring/dynatrace/).
+
+2. Set up a custom alert rule to be notified if the 90th percentile of the response time of your service is slower than expected. This custom alert rule does very much depend on your [SLOs](https://github.com/keptn/spec/blob/master/sre.md). 
+
+    An example can be seen in this screenshot:
+    ![Custom alert](./assets/dt-custom-alert.png)
+
+    Please note that in this case, the title of the custom alert has to match the remediation name in the remediation.yaml file, i.e., in this case `responsetime_p90_sockshop_carts`.
```

Comment:
OK, now we're getting to the bottom of this. To me, this seems very bespoke and not generic enough to map alert title to remediation action. While this might be OK for the first version, we need to think of a better way to do that.

Reply:
Correct, it is not a generic solution, but more a pragmatic one to directly have the violated SLO as part of the alert details (in this case the problem title) as well as in the name of the remediation.